tests/functional/codegen/features/test_reverting.py

```diff
@@ -17,14 +17,11 @@ def foo():
 
     revert_bytes = method_id("NoFives()")
 
-    assert_tx_failed(
-        lambda: get_contract_with_gas_estimation(reverty_code).foo(transact={}),
-        TransactionFailed,
-        exc_text=f"execution reverted: {revert_bytes}",
-    )
+    with tx_failed(TransactionFailed, exc_text=f"execution reverted: {revert_bytes}"):
+        get_contract_with_gas_estimation(reverty_code).foo(transact={})
```

Comment:
this would be better written over multiple lines --
```python
c = get_contract(reverty_code)
with tx_failed(exc_text=...):
    c.foo(transact={})
```